docs-src/extensionFunctions.adoc

```diff
@@ -0,0 +1,20 @@
+=== Extension functions
+
+Extension functions allow us to attach custom functions to any library or custom code we write. You can extend String, Collection or 
+any other known type with some custom behavior. Given example creates `isContainedBy` extension function that we can directly call on 
+any kind of `T` type, as though it was part of the standard library.
```

Comment:
Well, if you like - we can have basic and advanced categories of examples, so that this one, for example, might end up in advanced section. 
I'd rather go that way. 
Meanwhile - I'll prepare some basic case.